frontend/src/v5/ui/routes/viewer/tickets/commentsPanel/comment/comment.component.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { CurrentUserHooksSelectors } from '@/v5/services/selectorsHooks';
+import { IComment } from '@/v5/store/tickets/tickets.types';
+import { useEffect, useState } from 'react';
+import { extractMessage, extractMetadata, getRelativeTime } from './comment.helpers';
+import { CurrentUserComment } from './currentUserComment/currentUserComment.component';
+import { OtherUserComment } from './otherUserComment/otherUserComment.component';
+
+export type CommentProps = IComment & {
+	onDelete: (commentId) => void;
+	onReply: (commentId) => void;
+	onEdit: (commentId, newMessage: string) => void;
+};
+export const Comment = ({ updatedAt, author, message, deleted, _id, ...props }: CommentProps) => {
+	const [commentAge, setCommentAge] = useState(getRelativeTime(updatedAt));
+
+	const isCurrentUser = CurrentUserHooksSelectors.selectUsername() === author;
+	const metadata = extractMetadata(message);
+	const noMetadataMessage = !deleted ? extractMessage(message) : message;
+
+	const updateMessageAge = () => setCommentAge(getRelativeTime(updatedAt));
+
+	useEffect(() => {
+		const intervalId = window.setInterval(updateMessageAge, 10_000);
+		return () => {
+			updateMessageAge();
+			clearInterval(intervalId);
+		};
+	}, [updateMessageAge]);
```

Comment:
This useEffect seems to be being called on every comment when just typing in the comment input box. The result is that it gets very slow to type when a comment area has 30+ comments.

This seems to be happening because the props on all the comments change
![image](https://user-images.githubusercontent.com/62481250/216957006-6105ae5d-2c49-41df-98ac-c9952bccdc5e.png)

Santiago won't like this, but you might need to do some memoising and useCallback so that the comment components don't needlessly rerender, although they might be other solutions such as adding keys and other react functions that will appease him

Reply:
I'm still getting the problem where all the comments rerender when typing

Reply:
Fixed